File: content/search/source/includes/index/field-types/autocomplete/code-snippets/cpp/create_index_email_example.cpp
#include <iostream>

#include <bsoncxx/builder/basic/document.hpp>
#include <bsoncxx/json.hpp>
#include <mongocxx/client.hpp>
#include <mongocxx/instance.hpp>
#include <mongocxx/uri.hpp>
#include <mongocxx/search_index.hpp>

using bsoncxx::builder::basic::kvp;
using bsoncxx::builder::basic::make_document;

int main(){

    mongocxx::instance instance;
    mongocxx::uri uri("<connection-string>");
    mongocxx::client client(uri);

    auto db = client["sample_mflix"];
    auto collection = db["users"];

    // instantiate a ``mongocxx::search_index_view`` on your collection
    auto siv = collection.search_indexes();

    {
        auto name = "default";
        auto definition = make_document(
            kvp("mappings", make_document(
                kvp("dynamic", true),
                kvp("fields", make_document(
                    kvp("email", make_document(
                        kvp("type", "autocomplete"),
                        kvp("analyzer", "lucene.keyword"),
                        kvp("tokenization", "nGram"),
                        kvp("minGrams", 3),
                        kvp("maxGrams", 15),
                        kvp("foldDiacritics", false)
                    ))
                ))
            ))
        );

        auto model = mongocxx::search_index_model(name, definition.view());
        // Create the search index
        auto result = siv.create_one(model);
        if (result) {
            std::cout << "New index name: " << *result << std::endl;
        } else {
            std::cout << "Failed to create search index." << std::endl;
        }
    }
}
}

File: content/search/source/includes/index/field-types/autocomplete/code-snippets/java/create_index_email_example.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class CreateIndex {
    public static void main(String[] args) {
        // connect to your Atlas cluster
        String uri = "<connection-string>";

        try (MongoClient mongoClient = MongoClients.create(uri)) {
            // set namespace
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("users");
            String indexName = "default";

            Document searchIdx = new Document("mappings",
                    new Document("dynamic", true)
                        .append("fields", new Document("email", 
                                              new Document("type", "autocomplete")
                                                  .append("analyzer", "lucene.keyword")
                                                  .append("tokenization", "nGram")
                                                  .append("minGrams", 3)
                                                  .append("maxGrams", 15)
                                                  .append("foldDiacritics", false)
                                                  )));

            collection.createSearchIndex(indexName, searchIdx);
            System.out.println("New index name: " + indexName);
        } catch (Exception e) {
            System.err.println("Error creating Atlas Search index: " + e.getMessage());
        }
    }
}